Adds email and password validations and updates TextField Component

hannahbergam opened the pull request:
This PR handles several small updates:

1. Adds email validation for both student and parent emails
2. Obscures the passwords with the dots instead of showing them onscreen
3. Refactors the TextField to enable the type=password necessary for ^
4. Updates the test file to look for new validation messages
5. Adds confirm password validation to make sure the second password matches the first

Videos of both flows:

https://github.com/user-attachments/assets/a9d77218-3ef0-4b56-8d53-a45a5f59f9b5


https://github.com/user-attachments/assets/18b4c691-e881-446a-aef6-de867b22692a




## Links

- spec: []()
- jira ticket: https://codedotorg.atlassian.net/browse/ACQ-2287

## Testing story

<!--
  Does your change include appropriate tests?
  If so, please describe how the tests included in this PR are sufficient.
  If not, please explain why this change does not need to be tested.
-->

<!-- Other aspects to consider. Delete any sections that are not relevant to your change. -->

## Deployment strategy

## Follow-up work

<!--
  List (ideally with Jira links) any clean-up or technical debt that will be addressed in future work.
-->

## Privacy

<!--
  1.	Does this change involve the collection, use, or sharing of new Personal Data?
  6.	Does this change involve a new or changed use or sharing of existing Personal Data?
-->

## Security

<!-- Link to Jira task(s) where sensitive security issues are discussed privately. -->

## Caching

## PR Checklist:

<!--
  The final step! Before you create your PR, double-check that everything is in order.
  Change [ ] to [X] during creation to check boxes.
-->

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

levadadenys commented on apps/src/componentLibrary/textField/TextField.tsx:
```diff
@@ -21,6 +21,8 @@ export interface TextFieldProps extends AriaAttributes {
   /** The value attribute specifies the value of an input element. */
   value?: string;
+  /** Can be specified password if needed. */
+  type?: string;
```
I think it's better to rename this prop into `inputType` since `type` might lead to some confusions/misunderstading in the future, giving that in other DSCO components we're using `type` prop mostly for styles related logic

levadadenys replied:
Also, I do think we need to specify all possible `inputType` values here, cause there're a lot of input types in html and not all of them will work here. `'text' | 'email' | 'password' | 'number'` should work I think, please let me know if there're any other `inputTypes` we might need for this component that I've missed.

hannahbergam replied:
I removed the updates from this PR and will tag you in the new one!

levadadenys reviewed (changes requested):
Thanks a lot for adding this functionality to `TextField`!
Though, I think it's better to move `TextField` update to a separate PR giving our DSCO workflow. Also along with this update we need to update `Changelog`, add stories describing possible `inputType` options/usages

TurnerRiley reviewed (approved):
Love these updates!! I'll defer to Denys for the design component updates

hannahbergam commented:
> Love these updates!! I'll defer to Denys for the design component updates

Just an fyi, I removed the design system updates which meant I can't obscure the password yet with this PR. I added comments inline of what I will update next, but the rest of the work can stay!

levadadenys reviewed (approved):
Thanks a lot for splitting DSCO TextTfield into separate PR! Approved https://github.com/code-dot-org/code-dot-org/pull/60852, approving this one as well!
